File: Java/src/com/algorithms/week4/MajorityProblem.java
package com.algorithms.week4;

import java.util.Arrays;
import java.util.Scanner;

public class MajorityProblem {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int n = scan.nextInt();
		int[] arr = new int[n];
		for (int i = 0; i < n; i++)
			arr[i] = scan.nextInt();

		majorityElement(arr);
		scan.close();
	}

	public static void majorityElement(int[] arr) {
		Arrays.sort(arr);
		int count = 1, max_ele = -1, temp = arr[0], f = 0;

		for (int i = 1; i < arr.length; i++) {
			if (temp == arr[i]) {
				count++;
			} else {
				count = 1;
				temp = arr[i];
			}

			if (max_ele < count) {
				max_ele = count;
				if (max_ele > (arr.length / 2)) {
					f = 1;
					break;
				}
			}
		}
		System.out.println(f);
	}

}

File: C++/week4/MajorityProblem.cpp
#include<bits/stdc++.h>

using namespace std;

void majorityElement(int* arr, int n) {
    sort(arr, arr + n);
    int count = 1, max_ele = -1, temp = arr[0], f = 0;

    for (int i = 1; i < n; i++) {
        if (temp == arr[i]) {
            count++;
        } else {
            count = 1;
            temp = arr[i];
        }

        if (max_ele < count) {
            max_ele = count;
            if (max_ele > (n / 2)) {
                f = 1;
                break;
            }
        }
    }
    cout<<f<<endl;
}

int main() {
    int n;
    cin>>n;
    int* arr = new int[n];
    for (int i = 0; i < n; i++)
        cin>>arr[i];

    majorityElement(arr, n);
}
